Feature: Reopen last storage command

aphased opened the pull request:
## What

This PR adds three new sub-commands as a (configless/always-on) feature. Example: `/bp -` (the single dash). Executing it will re-open the backpack that was last opened using `/bp [int]`.  
The inspiration for this comes from Unix-style `cd -` for going to the previous current directory. Also, I'm using `/bp` and `/ec` for brevity everywhere, `/backpack` and `/enderchest` work as well of course.

The new command `/shlo` (and `/shlastopened`) will always open the last-used storage irrespective of whether it was an ender chest or backpack page.

The key/command combinations are not yet taken AFAIK.
Currently in an unmodded client, executing
- `/bp -` gives `Command Failed: Non-Numeric argument!`.
- `/ec -` opens the first ender chest page, which is also still reachable via `/ec` or `/ec 1`
- `/shlastopened` (and its short-form variant `/shlo`) are of course also still unregistered

---

Apart from typing it out as a command (or repeatedly accessing it via the up arrow), this feature also has the advantage that a **single hotkey** from any number of mods may be defined and used the same way every time you need to access _any_ storage page multiple times in a row for removing/adding some items to it.

Also potentially noteworthy regarding the implementation, I made the explicit decision against resetting the most recent backpack/ender chest slot value in case of a non-integer argument. In this case, it will just keep the previous valid integer backpack number "active".

This feature was first suggested [here](https://discord.com/channels/997079228510117908/1258925643828035744).


## Changelog New Features
+ Added commands "/bp -", "/ec -" and "/shlastopened" to reopen last opened storage pages. - aphased

aphased committed:
Add `/bp -` command

qtlunya commented:
Cool idea, can we have the same thing for ender chest as well?

aphased committed:
Fix detekt linting issue

Xclipsen commented:
Thank you for your hard work on this feature! I really hope it gets accepted, as it would make me genuinely happy to use it actively. I believe it has a lot of potential, and I appreciate the effort you’ve put into it.

CalMWolfs committed:
Merge branch 'beta' into beta

aphased commented:
Converting to draft since I would still like to add @lunaynx's [suggestion](https://github.com/hannibal002/SkyHanni/pull/2900#issuecomment-2460667087) (see above) first.

aphased committed:
Add support for opening `ec` and either storage

The newly-added `/lo` will always open the last-used storage irrespective of whether it was an ender chest or backpack page.

aphased committed:
Fix not cancelling event & trigger on invalid page

(`/ec -1` works Hypixel-side, but not in the `/ec -` command logic)

aphased committed:
Fix triggering `/lo` on `/locraw`

(oops)

aphased committed:
Fix not updating the last storage value after use

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/commands/OpenLastStorage.kt:
```diff
@@ -0,0 +1,125 @@
+        }
+    }
+
+    private fun handleBackpack(event: MessageSendToServerEvent): Boolean {
```
the event is only ued for reading the message. instead of using the event here, you can directly give the function only  message

aphased replied:
Not entirely sure, but it feels a bit unavoidable to pass it in because I need to set the event to `isCanceled`

CalMWolfs replied:
Resolved in my commit

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/commands/OpenLastStorage.kt:
```diff
@@ -0,0 +1,125 @@
+            return false
+        }
+
+        if (event.message.startsWith("/backpack -", ignoreCase = true) ||
```
instead of ignore case everywhere, you can make the message to lower case once at the beginning

aphased replied:
Done

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/commands/OpenLastStorage.kt:
```diff
@@ -0,0 +1,125 @@
+    }
+
+    private var lastBackpack = -1
+    private var lastEnderChest = -1
```
please use kotlin's null instead of -1 as default/invalid vlaue

aphased replied:
Done

hannibal002 reviewed (changes requested):
as you can see in this comparison, the two functions handleBackpack and handleEnderChest are doing the same thing, just with different values. please try using only one function for both
![image](https://github.com/user-attachments/assets/e24f9d12-e0f6-4929-b071-40a481dffbd0)

aphased committed:
Use null over -1 & combine functionality where same

This drastically reduces repetitiveness.
Also, crucially, fixed not updating the last stored _type_ of storage (backpack vs. ender chest), only the page number/index; and this time in the correct location.
Also trimmed use of `event.message.startsWith(x, ignoreCase = true)` where possible.

aphased committed:
Fix detekt linting issues (partially)

aphased committed:
Fix detekt (accidentally ate a newline)

aphased committed:
Use CommandRegistrationEvent for new command

As advised in the #coding channel on Discord.
Also renamed both new commands to be prefixed with "sh"
(`/shlastopened`, `/shlo`)

aphased committed:
Assign new command a category

(`/shlastopened` is a normal user command)

CalMWolfs committed:
Merge branch 'refs/heads/beta' into fork/aphased/beta

CalMWolfs committed:
detekt is silly

aphased committed:
Fix last detekt issue

I think 🙏

CalMWolfs committed:
code cleanup

CalMWolfs committed:
further code cleanup

CalMWolfs committed:
baseline

CalMWolfs reviewed (approved):
Worked for me

CalMWolfs committed:
Merge branch 'beta' into beta

CalMWolfs committed:
fix merge

hannibal002 committed:
code cleanup and using formatIntOrUserError

hannibal002 reviewed (changes requested):
Please add a toggle to disable this behaviour.

CalMWolfs committed:
add config option

CalMWolfs committed:
Merge branch 'beta' into beta

hannibal002 committed:
added clickable
